src/WordPress/JsonMapper/Evaluators/NamespaceResolver.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+namespace WordPress\JsonMapper\Evaluators;
+
+use PhpParser\NodeTraverser;
+use PhpParser\ParserFactory;
+use WordPress\JsonMapper\Import;
+use WordPress\JsonMapper\JsonMapper;
+use WordPress\JsonMapper\ObjectWrapper;
+use WordPress\JsonMapper\Property\Property;
+use WordPress\JsonMapper\Property\PropertyMap;
+use WordPress\JsonMapper\Property\PropertyType;
+use WordPress\JsonMapper\UseNodeVisitor;
+
+class NamespaceResolver implements JsonEvaluatorInterface {
+
+	private $scalar_types = array( 'string', 'bool', 'boolean', 'int', 'integer', 'double', 'float' );
+
+	public function __construct() {}
+
+	public function evaluate(
+		\stdClass $json,
+		ObjectWrapper $object_wrapper,
+		PropertyMap $property_map,
+		JsonMapper $mapper
+	) {
+		foreach ( $this->fetchPropertyMapForObject( $object_wrapper, $property_map ) as $property ) {
+			$property_map->addProperty( $property );
+		}
+	}
+
+	private function fetchPropertyMapForObject( ObjectWrapper $object, PropertyMap $originalPropertyMap ): PropertyMap {
+		$intermediatePropertyMap = new PropertyMap();
+		$imports                 = self::getImports( $object->getReflectedObject() );
+
+		/** @var Property $property */
+		foreach ( $originalPropertyMap as $property ) {
+			$types = $property->get_property_types();
+			foreach ( $types as $index => $type ) {
+				$types[ $index ] = $this->resolveSingleType( $type, $object, $imports );
+			}
+			$intermediatePropertyMap->addProperty( $property->as_builder()->setTypes( ...$types )->build() );
+		}
+
+		return $intermediatePropertyMap;
+	}
+
+	/** @return Import[] */
+	private static function getImports( \ReflectionClass $class ): array {
+		if ( ! $class->isUserDefined() ) {
+			return array();
+		}
+
+		$filename = $class->getFileName();
+		if ( $filename === false || \substr( $filename, -13 ) === "eval()'d code" ) {
+			throw new \RuntimeException( "Class {$class->getName()} has no filename available" );
+		}
+
+		if ( $class->getParentClass() === false ) {
+			return self::getImportsForFileName( $filename );
+		}
+
+		return array_unique(
+			array_merge( self::getImportsForFileName( $filename ), self::getImports( $class->getParentClass() ) ),
+			SORT_REGULAR
+		);
+	}
+
+	/** @return Import[] */
+	private static function getImportsForFileName( string $filename ): array {
```

Comment:
Note for later: This might be problematic in eval()-like context where there's no file to parse. I don't think there's any other way to parse those `use` statements, though, so perhaps that's a limitation of the `BlueprintMapper` to document in a doc block above that class.

Reply:
I would like to take a look at those imports in a subsequent PR. For now - what type o message do you propose?

Reply:
Gone. Imports do not seem relevant for the mapper at this point in time.